apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessment/assessment.ts

```diff
@@ -24,7 +34,62 @@ router.get(
       ManualGradingQuestionSchema,
     );
     const num_open_instances = questions[0]?.num_open_instances || 0;
-    res.send(ManualGradingAssessment({ resLocals: res.locals, questions, num_open_instances }));
+    const courseStaff = await selectUsersWithCourseInstanceAccess({
+      course_instance_id: res.locals.course_instance.id,
+    });
+    res.send(
+      ManualGradingAssessment({
+        resLocals: res.locals,
+        questions,
+        courseStaff,
+        num_open_instances,
+      }),
+    );
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_edit) {
+      throw new HttpStatusError(403, 'Access denied (must be a student data editor)');
+    }
+    if (req.body.__action === 'assign_graders') {
+      if (!req.body.assigned_grader) {
+        flash('error', 'No graders were selected for assignment.');
+        res.redirect(req.originalUrl);
+        return;
+      }
+      const graderIds: string[] = Array.isArray(req.body.assigned_grader)
+        ? req.body.assigned_grader
+        : [req.body.assigned_grader];
```

Comment:
AFAICT we're completely lacking any validation that these users are student data editors within this course. Did I miss that somewhere?

Reply:
Indeed it was missing. Done in 31807dbc4.